src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp

```diff
@@ -203,6 +203,7 @@ struct RemoveManyObjectStorageOperation final : public IDiskObjectStorageOperati
                 if (unlink_outcome && !keep_all_batch_data && !file_names_remove_metadata_only.contains(fs::path(path).filename()))
                 {
                     objects_to_remove.emplace_back(ObjectsToRemove{std::move(objects), std::move(unlink_outcome)});
+                    removed_files.push_back(path);
```

Comment:
The `removed_files` is a bit confusing. Moreover, we already have `remove_paths` in the same scope

Reply:
Ok. Will change this naming.